pangolin/utils/initialising.py

```diff
@@ -13,6 +13,15 @@
 from pangolin import __version__
 
 import pangolin_data
+pangolin_assignment_version = None
+pangolin_assignment_path = None
+try:
+    import pangolin_assignment
+    pangolin_assignment_version = pangolin_assignment.__version__
+    pangolin_assignment_path = pangolin_assignment.__path__[0]
+except ImportError:
+    # if we can't import the module, leave the variables as None
+    pass
```

Comment:
```suggestion
class PangolinAssignmentWrapper():
    __version__ = None
    __path__ = [None]

pangolin_assignment = PangolinAssignmentWrapper()
try:
    import pangolin_assignment
except ImportError:
    # if we can't import the module, we replace it with a mock with suitable attributes
    pangolin_assignment = PangolinAssignmentWrapper()
```
With this trick you'll be able to assume a module-like object everywhere in this code. That means you can get rid of the `global` declarations later on and treat this modules just like the others that you know must exist.